src/com/linkedlistsample/LinkedListSample.java

```diff
@@ -0,0 +1,133 @@
+package com.linkedlistsample;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Stack;
+
+public class LinkedListSample {
+	public static void main(String args[]) {
+		LinkedList<Integer> list1 = new LinkedList<>();
+		LinkedList<Integer> list2 = new LinkedList<>();
+		list1.add(3);
+		list1.add(4);
+		list1.add(2);
+		list2.add(4);
+		list2.add(6);
+		list2.add(5);
+
+		try {
+			LinkedList<Integer> list3 = findSum1(list1, list2);
+			printList(list3);
+			LinkedList<Integer> list4 = findSum2(list1, list2);
+			printList(list4);
+			LinkedList<Integer> list5 = findSum3(list1, list2);
+			printList(list5);
+		} catch (Exception e) {
+			System.out.println("no output");
+		}
+
+	}
+
+	/*
+	 * TC:0(n^2),SC:0(n^2) without using stacks fetching through the entire list all
+	 * the time
+	 */
+	private static LinkedList<Integer> findSum1(LinkedList<Integer> list1, LinkedList<Integer> list2)
+			throws NullPointerException {
+		if (list1.isEmpty() || list2.isEmpty() || list1.size() != list2.size()) {
+			throw new NullPointerException();
+
+		} else {
+			int div = 0;
+			LinkedList<Integer> list3 = new LinkedList<>();
+			for (int i = list1.size(), j = list2.size(); i >= 1 || j >= 1; i--, j--) {
+				list3.add((div + list1.get(i - 1) + list2.get(j - 1)) % 10);
+				div = (div + list1.get(i - 1) + list2.get(j - 1)) / 10;
+			}
+			if (div != 0) {
+				list3.add(div);
+			}
+			return list3;
+		}
+
+	}
+
+	/*
+	 * TC:0(n),SC:0(n) using two stacks
+	 */
+	private static LinkedList<Integer> findSum2(LinkedList<Integer> list1, LinkedList<Integer> list2)
+			throws NullPointerException {
+		if (list1.isEmpty() || list2.isEmpty() || list1.size() != list2.size()) {
```

Comment:
Add a util method to check for null and empty
boolean isEmpty(List<String> list) {
   if (list == null || list.size() == 0) return true;
   return false;
}